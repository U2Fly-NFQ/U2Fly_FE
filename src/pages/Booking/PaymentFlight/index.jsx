import React, { useEffect, useState } from 'react'
import './index.scss'
import { Form, Radio } from 'antd'
import { ButtonOfPage } from '../../../components'
import logoStripe from '../../../../src/assets/images/system/stripe-icon.jpg'
import { useDispatch, useSelector } from 'react-redux'
import {
  getDiscountForBookingAirline,
  getInfoFlightInBookingFight,
  getInfoFlightInBookingSeat,
  getInfoPriceAfterDiscount,
  getRoundTripBookingFlight,
  getUserInformation,
} from '../../../redux/selectors'
import { createBookingFlight } from '../../../redux/slices/bookingFlightsSlice'
import { useTranslation } from 'react-i18next'
export default function PaymentFlight() {
  const [value, setValue] = useState(1)
  const [dataBooking, setDataBooking] = useState()
  const priceDiscount = useSelector(getInfoPriceAfterDiscount)
  const getDiscountInfo = useSelector(getDiscountForBookingAirline)
  const getPrice = useSelector(getInfoFlightInBookingSeat)
  const getFlightData = useSelector(getInfoFlightInBookingFight)
  const getSeatData = useSelector(getInfoFlightInBookingSeat)
  const userInformation = useSelector(getUserInformation)
  const getRoundTrip = useSelector(getRoundTripBookingFlight)
  const flightLocal = JSON.parse(localStorage.getItem('flight'))
  const dispatch = useDispatch()
  const { t } = useTranslation()

  const onChange = (e) => {
    setValue(e.target.value)
  }
  useEffect(() => {
    setDataBooking(dataPayment[value])
  }, [value])
  const onFinish = () => {
    let priceTotal =
      getRoundTrip.seat !== undefined
        ? getPrice.price * 110 + getRoundTrip.seat.price * 110
        : getPrice.price * 110
    let discount =
      (getRoundTrip.seat !== undefined
        ? getPrice.price + getRoundTrip.seat.price
        : getPrice.price) * getDiscountInfo.percent
    let fetchDataValue = {
      passengerId: userInformation.id,
      flightId: getRoundTrip.id
        ? `${getFlightData.id},${getRoundTrip.id}`
        : `${getFlightData.id}`,
      seatTypeId: getSeatData.id,
      totalPrice: Math.floor(
        (priceTotal - discount * 100) * flightLocal.seatAvailable
      ),
      discountId: getDiscountInfo.id || 1,
      ticketOwner: userInformation.firstName,
    }
    dispatch(createBookingFlight(fetchDataValue))
  }
  let dataPayment = [
    {
      paymentMethod: 'Stripe',
      logo: logoStripe,
      render: (
        <>
          <Form.Item style={{ marginTop: '20px' }}>
            <ButtonOfPage title={t('flight-booking-page.Pay Now')} />
          </Form.Item>
        </>
      ),
    },
  ]

  return (
    <>
      <div className="booking-page__container__item__title">
        <h2>Payment Method</h2>
      </div>
      <div style={{ width: '100%' }}>
        <Radio.Group
          onChange={onChange}
          value={value}
          style={{ width: '100%' }}
          className="paymnet-button__radio"
        >
          {dataPayment &&
            dataPayment.map((item, index) => {
              return (
                <div className="paymnet-button__radio form-control">
                  <Radio
                    value={index}
                    key={index}
                    className="paymnet-button__radio__btn"
                  >
                    {item.paymentMethod}
                    <div className="paymnet-button__radio__logo ">
                      <img src={item.logo} alt="stripe-logo" />
                    </div>
                  </Radio>
                </div>
              )
            })}
        </Radio.Group>
        <div className="BookingPayment-content">
          <Form onFinish={onFinish}>{dataBooking && dataBooking.render}</Form>
        </div>
      </div>
    </>
  )
}
