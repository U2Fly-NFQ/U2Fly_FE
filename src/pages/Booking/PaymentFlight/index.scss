.paymnet-button__radio {
  display: flex;
  gap: 0.5rem;
  flex-direction: column;
  justify-content: center;
  height: fit-content;
  padding: 5px;
  &__btn {
    span {
      font-size: 17px;
      font-weight: 500;
      display: flex;
      align-items: center;
      &:last-child {
        flex: 1;
        justify-content: space-between;
      }
    }
  }
  &__logo {
    img {
      width: 100px;
    }
  }
}
