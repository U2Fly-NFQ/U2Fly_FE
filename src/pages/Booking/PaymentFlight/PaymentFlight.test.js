import React, { Suspense } from 'react'
import { I18nextProvider } from 'react-i18next'
import renderer from 'react-test-renderer'
import i18n from '../../../translations'
import configureStore from 'redux-mock-store'
import { Provider } from 'react-redux'
import { fireEvent, render, screen } from '@testing-library/react'
import { BrowserRouter as Router } from 'react-router-dom'
import PaymentFlight from './'

const mockStore = configureStore([])

describe('Booking Passenger bar test', () => {
  let store
  let component
  beforeEach(() => {
    store = mockStore({
      bookingFlight: {
        userInformation: { id: '123', name: 'John Smith' },
        dataFlight: { seat: { price: 56, name: '50' } },
        discountInfo: { percent: 0 },
        dataRoundTripFlight: { seat: { price: 56, name: '50' } },
      },
    })
  })
  it('renders correctly', () => {
    const tree = renderer
      .create(
        <Provider store={store}>
          <I18nextProvider i18n={i18n}>
            <Router>
              <Suspense>
                <PaymentFlight />
              </Suspense>
            </Router>
          </I18nextProvider>
        </Provider>
      )
      .toJSON()
    expect(tree).toMatchSnapshot()
  })
  it('should subtmit correctly', async () => {
    render(
      <Provider store={store}>
        <I18nextProvider i18n={i18n}>
          <Router>
            <Suspense>
              <PaymentFlight />
            </Suspense>
          </Router>
        </I18nextProvider>
      </Provider>
    )

    const radioButton = screen.getByText('Stripe')
    fireEvent.click(radioButton)
    const Payment = await screen.findByTestId('formPayment')
    fireEvent.submit(Payment)
  })
})
