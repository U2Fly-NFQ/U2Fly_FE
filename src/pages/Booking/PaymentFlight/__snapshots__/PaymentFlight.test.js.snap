// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Booking Passenger bar test renders correctly 1`] = `
Array [
  <div
    className="booking-page__container__item__title"
  >
    <h2>
      Payment Method
    </h2>
  </div>,
  <div
    style={
      Object {
        "width": "100%",
      }
    }
  >
    <div
      className="ant-radio-group ant-radio-group-outline paymnet-button__radio"
      style={
        Object {
          "width": "100%",
        }
      }
    >
      <div
        className="paymnet-button__radio form-control"
      >
        <label
          className="ant-radio-wrapper paymnet-button__radio__btn"
        >
          <span
            className="ant-radio"
            style={Object {}}
          >
            <input
              checked={false}
              className="ant-radio-input"
              onBlur={[Function]}
              onChange={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              onKeyPress={[Function]}
              onKeyUp={[Function]}
              type="radio"
              value={0}
            />
            <span
              className="ant-radio-inner"
            />
          </span>
          <span>
            Stripe
            <div
              className="paymnet-button__radio__logo "
            >
              <img
                alt="stripe-logo"
                src="stripe-icon.jpg"
              />
            </div>
          </span>
        </label>
      </div>
    </div>
    <div
      className="BookingPayment-content"
    >
      <form
        className="ant-form ant-form-horizontal"
        onReset={[Function]}
        onSubmit={[Function]}
        test-dataid="formPayment"
      />
    </div>
  </div>,
]
`;
